src/App.tsx

```diff
@@ -1,26 +1,346 @@
-
-const USER_ID = 0;
+import {
+  deleteTodo,
+  getTodos,
+  postTodo,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './Header';
+import { TodoElem } from './TodoElem';
+import { Footer } from './Footer';
+import { ErrorComponent } from './Error';
+import { TransitionGroup } from 'react-transition-group';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [value, setValue] = useState<string>('');
+  const [selectedLink, setSelectedLink] = useState<string>('all');
+  const [editingId, setEditingId] = useState<number | null>(null);
+  const [titleForEditing, setTitleForEditing] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState<string>('');
+  const [savingIs, setSavingIds] = useState<number[]>([]);
+
+  const titleField = useRef<HTMLInputElement>(null);
+  const editInputRef = useRef<HTMLInputElement>(null);
+
+  function hideError() {
+    setTimeout(() => setError(''), 3000);
+  }
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(result => {
+        setTodos(result);
+      })
+      .catch(() => {
+        setError('Unable to load todos');
+        hideError();
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (!isLoading && titleField.current) {
+      titleField.current.focus();
+    }
+  }, [isLoading]);
+
+  useEffect(() => {
+    if (editingId !== null) {
+      const editingTodo = todos.find(todo => todo.id === editingId);
+
+      if (editingTodo) {
+        setTitleForEditing(editingTodo.title);
+      }
+
+      editInputRef.current?.focus();
+    }
+  }, [editingId, todos, isLoading]);
+
+  function applyFilter(filter: string, source: Todo[]) {
+    if (filter === 'completed') {
+      setVisibleTodos(source.filter(x => x.completed));
+    } else if (filter === 'active') {
+      setVisibleTodos(source.filter(x => !x.completed));
+    } else {
+      setVisibleTodos(source);
+    }
+  }
+
+  useEffect(() => {
+    applyFilter(selectedLink, todos);
+  }, [selectedLink, todos]);
+
+  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    setError('');
+    setIsLoading(true);
+
+    const maxId = Math.max(...todos.map(todo => todo.id)) + 1;
+    const trimmed = value.trim();
+
+    setSavingIds([maxId]);
+
+    if (!trimmed) {
+      setError('Title should not be empty');
+      hideError();
+      setIsLoading(false);
+      setSavingIds([]);
+
+      return;
+    }
+
+    setTempTodo({
+      title: trimmed,
+      id: 0,
+      userId: USER_ID,
+      completed: false,
+    });
+
+    const data = {
+      title: trimmed,
+      userId: USER_ID,
+      completed: false,
+      id: maxId,
+    };
```

Comment:
```suggestion
    const data = {
      title: trimmed,
      userId: USER_ID,
      completed: false,
    };
```
id will be added on Server, and you'll receive a Response with fully-prepared TodoItem.
Check `api/todos.ts` with add Todo method